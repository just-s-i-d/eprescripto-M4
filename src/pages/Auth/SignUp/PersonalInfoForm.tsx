import { Form, Input, Select } from "antd";

const PersonalInfoForm = () => {
  return (
    <>
      <Form.Item
        label="Name"
        name="docName"
        className="text-xl"
        required
        rules={[
          {
            pattern: /^(Dr\.?|Doctor)?\s?[A-Za-z\s.'-]+$/,
            message: "Special characters not allowed",
          },
          {
            required: true,
            message: "Field cannot be empty",
          },
        ]}
      >
        <Input placeholder="ex. John Doe or Dr. John Doe" />
      </Form.Item>
      <Form.Item
        label="Email"
        name="email"
        required
        rules={[
          {
            required: true,
            message: "Field cannot be empty",
          },
          {
            pattern: /^[\w.-]+@[\w.-]+\.\w+$/,
            message: "Enter a valid mail",
          },
        ]}
      >
        <Input placeholder="ex. John@example.com" />
      </Form.Item>
      <div className="w-full flex justify-between">
        <Form.Item
          label="Gender"
          name="gender"
          required
          className="w-[45%]"
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
          ]}
        >
          <Select
            placeholder="Select a gender"
            options={[
              { value: "male", label: "Male" },
              { value: "female", label: "Female" },
              { value: "other", label: "Other" },
            ]}
          />
        </Form.Item>
        <Form.Item
          label="Date of Birth"
          required
          className="w-[45%]"
          name="dob"
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
          ]}
        >
          <Input type="date" placeholder="ex. 24" />
        </Form.Item>
      </div>
    </>
  );
};

export default PersonalInfoForm;
