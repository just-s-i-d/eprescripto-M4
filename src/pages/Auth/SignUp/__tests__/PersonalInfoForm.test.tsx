import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@test/matchMedia";
import PersonalInfoForm from "../PersonalInfoForm";

describe("Personal form tests", () => {
  it("Render the form fields correctly", () => {
    const { asFragment } = render(<PersonalInfoForm />);
    expect(screen.getByLabelText("Name")).toBeInTheDocument();
    expect(screen.getByLabelText("Email")).toBeInTheDocument();
    expect(screen.getByLabelText("Gender")).toBeInTheDocument();
    expect(screen.getByLabelText("Date of Birth")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
