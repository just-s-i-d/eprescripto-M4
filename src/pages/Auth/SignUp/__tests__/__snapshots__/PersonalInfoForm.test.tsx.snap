// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal form tests Render the form fields correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-form-item text-xl css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
      >
        <label
          class="ant-form-item-required"
          for="docName"
          title="Name"
        >
          Name
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              aria-required="true"
              class="ant-input css-dev-only-do-not-override-1adbn6x"
              id="docName"
              placeholder="ex. John Doe or Dr. John Doe"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-form-item css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
      >
        <label
          class="ant-form-item-required"
          for="email"
          title="Email"
        >
          Email
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              aria-required="true"
              class="ant-input css-dev-only-do-not-override-1adbn6x"
              id="email"
              placeholder="ex. John@example.com"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="w-full flex justify-between"
  >
    <div
      class="ant-form-item w-[45%] css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
        >
          <label
            class="ant-form-item-required"
            for="gender"
            title="Gender"
          >
            Gender
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow"
                data-testid="gender-select-signup"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="gender_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="gender_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="gender"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    Select a gender
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item w-[45%] css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
        >
          <label
            class="ant-form-item-required"
            for="dob"
            title="Date of Birth"
          >
            Date of Birth
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input css-dev-only-do-not-override-1adbn6x"
                data-testid="date-signup"
                id="dob"
                placeholder="ex. 24"
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
